### PR TITLE
Allow some time for the configuration propagation

### PR DESCRIPTION
**Pull request by irisdingbj**
Fix test flaky in https://github.com/istio/istio/issues/11882

### CODE REVIEW

File: tests/integration/pilot/security/authn_permissive_test.go [hunk 1]
```diff
@@ -119,22 +125,26 @@ func TestAuthnPermissive(t *testing.T) {
...
-	resp, err := pilot.CallDiscovery(req)
-	if err != nil {
-		t.Errorf("failed to call discovery %v", err)
-	}
-	for _, r := range resp.Resources {
-		foo := &xdsapi.Listener{}
-		if err := proto.UnmarshalAny(&r, foo); err != nil {
-			t.Errorf("failed to unmarshal %v", err)
+	endTime := time.Now().Add(timeout)
+	for {
```
Comment (nmittler) [inline tests/integration/pilot/security/authn_permissive_test.go]: Would you mind switching over to the retry library under pkg/test/util/retry?

Comment (ozevren) [reply tests/integration/pilot/security/authn_permissive_test.go]: +1

Comment (irisdingbj) [reply tests/integration/pilot/security/authn_permissive_test.go]: Thanks for the comments.  I have changed the logic to use WatchDiscovery instead to allow retry to get the response.

File: tests/integration/pilot/security/authn_permissive_test.go [hunk 2]
```diff
@@ -119,22 +121,26 @@ func TestAuthnPermissive(t *testing.T) {
...
 	}
-	for _, r := range resp.Resources {
-		foo := &xdsapi.Listener{}
-		if err := proto.UnmarshalAny(&r, foo); err != nil {
-			t.Errorf("failed to unmarshal %v", err)
-		}
-		if verifyListener(foo, t) {
-			return
-		}
+	err = pilot.WatchDiscovery(time.Second*10,
```
Comment (ozevren) [inline tests/integration/pilot/security/authn_permissive_test.go]: Note that 10 seconds may not be enough in the CI system.

**Pull request merged**